### PR TITLE
For CI, deploy to Arbitrum Sepolia instead of Base Sepolia

### PR DESCRIPTION
**Pull request by jplodge-pro**
Deploy to Arbitrum Sepolia testnet instead of Base Sepolia, for CI deployments.

### Done
- Changed `deploy:ci` build script to invoke `make deploy-arbitrumSepolia`
- Added Arbitrum Sepolia specific environment variables to the GitHub Action.
- Also updated the GitHub `testnet` Environment with same.
- Also updated the `foundry.toml` with same.
- Renamed the Arbitrum Sepolia specific make target to use the Network Alias
- Tweaked the `deploy` and `verify` commands in make to use the Build Scripts to assure a consistent state
- Added a means to use alternative Deployer Authentication, if needed.
    - For example, instead of a Private Key, we could use a `cast wallet` Account and Password, like so:
    ```DEPLOYER_AUTHENTICATION="--account DEVOPS_DEPLOYER --password xyz"```  
- Updated `testnet.toml` to reference the correct Chain Id.
- Pinned GitHub Action Foundry version (@lucasia)
- Changed to use Blockscout as the Verifier (@lucasia)

Resolves https://github.com/credbull/credbull-defi/issues/88

### CODE REVIEW

File: packages/contracts/Makefile [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
 verify:
-	@echo "Running verify '$(RPC_URL)' for '$(ENVIRONMENT)' environment."
-	rm -rf out && \
-	forge script script/DeployVaultFactory.s.sol:DeployVaultFactory --rpc-url $(RPC_URL) \
-		--private-key $(DEPLOYER_PRIVATE_KEY) --verify --resume -vvvv && \
-	yarn gen-types && \
+	@echo "Running verify of script '$(DEPLOY_SCRIPT)' against '$(RPC_URL)' for '$(ENVIRONMENT)' environment."
+	@yarn clean
+	@yarn build
+	forge script $(DEPLOY_SCRIPT) --rpc-url $(RPC_URL) $(DEPLOYER_AUTHENTICATION) --verify --resume -vvvv
```
Comment (lucasia) [inline packages/contracts/Makefile:42]: i think we need an "@" here as well.  this command is outputting the private key, which we want to suppress.

File: packages/contracts/package.json [hunk 2]
```diff
@@ -8,15 +8,15 @@
...
     "dev": "yarn rm-dbdata && anvil --config-out localhost.json & make deploy-local",
-    "build": "forge build && yarn gen-types",
+    "build": "forge build && yarn gen-types && tsc",
     "test": "forge test",
     "format": "forge fmt && prettier './script/**/*.js' --write",
     "lint": "forge fmt && eslint --fix --ignore-path .gitignore && yarn solhint './*(test|src)/**/*.sol'",
     "db-check": "tsc && node ./script/utils/checkDb.js",
     "db-export": "tsc && node ./script/utils/exporter.js",
     "deploy": "make deploy-local",
-    "deploy:ci": "yarn db-export && make deploy-baseSepolia",
```
Comment (lucasia) [inline packages/contracts/package.json:18]: this is a bit odd.  before we were calling db-export -> deploy -> db-export?

File: packages/contracts/package.json [hunk 3]
```diff
@@ -8,15 +8,15 @@
...
-    "gen-types": "yarn typechain --target ethers-v5 --out-dir ./types ./out/*.sol/*.json && tsc",
```
Comment (lucasia) [inline packages/contracts/package.json:19]: gen-types should come after deploy, shouldn't it?  if we remove the tsc here, we won't compile the types until another build call.

**Review by lucasia — CHANGES_REQUESTED** [review]
the deployment to arbitrumSepolia  from the branch works fine using the below on my machine at lest.  please confirm the versions of forge and forge-std you are using.  

**forge nightly (foundryup) forge 0.2.0 (4351742 2024-08-07T00:24:31.137910000Z)**
- `foundryup`
- ` forge --version`

**git submodule status**
 07263d193d621c4b2b0ce8b4d54af58f6957d97d forge-std (v1.9.1)
 dbb6104ce834628e473d2173bbc9d47f81a9eec3 openzeppelin-contracts (v5.0.0-12-gdbb6104c)

File: packages/contracts/package.json (hunk 2 above)
Comment (jplodge-pro) [reply packages/contracts/package.json:18]: Yes, it was a bit odd :)

File: packages/contracts/Makefile (hunk 1 above)
Comment (jplodge-pro) [reply packages/contracts/Makefile:42]: Agreed, done.

File: packages/contracts/Makefile [hunk 4]
```diff
@@ -1,43 +1,43 @@
...
+	yarn db-export
 
 verify:
-	@echo "Running verify '$(RPC_URL)' for '$(ENVIRONMENT)' environment."
-	rm -rf out && \
-	forge script script/DeployVaultFactory.s.sol:DeployVaultFactory --rpc-url $(RPC_URL) \
-		--private-key $(DEPLOYER_PRIVATE_KEY) --verify --resume -vvvv && \
-	yarn gen-types && \
+	@echo "Running verify of script '$(DEPLOY_SCRIPT)' against '$(RPC_URL)' for '$(ENVIRONMENT)' environment."
+	@yarn clean
```
[the remark this answers is not in the document]
Comment (jplodge-pro) [reply packages/contracts/Makefile:40]: For me, I would remove the `yarn clean` and `yarn build` entirely, given that `forge script` will compile any missing code anyway. I will restore the `rm -rf out`, though I think that is not quite correct (re-compiling `.sol` files to match the `broadcast` directory) as this is clearly a 'continue from where you left off' command.

File: packages/contracts/package.json (hunk 3 above)
Comment (jplodge-pro) [reply packages/contracts/package.json:19]: `gen-types` should come after a `forge build`. And yes, a `tsc` compiles all Typescript. However, having `tsc` in a 'generate code' command is not 'clean', to my mind. Convenient, yes, but not clean and not really necessary, as long as the TS compilation/transpilation always happens.
So, I changed the `build` script to include all 3 build-like tasks (compile, generate, transpile).

**Comment by jplodge-pro** [comment]
> the deployment to arbitrumSepolia from the branch works fine using the below on my machine at lest. please confirm the versions of forge and forge-std you are using.
> 
> **forge nightly (foundryup) forge 0.2.0 (4351742 2024-08-07T00:24:31.137910000Z)**
> 
> * `foundryup`
> * ` forge --version`
> 
> **git submodule status** 07263d193d621c4b2b0ce8b4d54af58f6957d97d forge-std (v1.9.1) dbb6104ce834628e473d2173bbc9d47f81a9eec3 openzeppelin-contracts (v5.0.0-12-gdbb6104c)

I foolishly ran `foundryup` before checking my version, so I don't know my previous version.
Now, however, my versions are as follows:
```$ forge --version
forge 0.2.0 (7f0f5b4 2024-08-08T00:19:27.650028841Z)
```
```$ git submodule status
 07263d193d621c4b2b0ce8b4d54af58f6957d97d lib/forge-std (v1.9.1)
 dbb6104ce834628e473d2173bbc9d47f81a9eec3 lib/openzeppelin-contracts (v5.0.0-12-gdbb6104c)
```
I will re-test the deployment.

**Comment by lucasia** [comment]
Closing - de-prioritized.

**Pull request closed without merging**